packages/core/src/command/feature-flags/ffs-on-dynamodb.ts

```diff
@@ -70,149 +60,148 @@ export const initialFeatureFlags: FeatureFlagDatastore = {
+  }
 
-export async function getFeatureFlagsRecord({
-  region,
-  tableName,
-  skipCache,
-}: {
-  region: string;
-  tableName: string;
-  skipCache?: boolean;
-}): Promise<FeatureFlagsRecord | undefined> {
-  const now = Date.now();
-  const age = now - (featureFlagsCache?.timestamp ?? 0);
-
-  if (!skipCache && featureFlagsCache && age < cacheDuration.asMilliseconds()) {
-    return featureFlagsCache.record;
+  /**
+   * Initialize the feature flags service.
+   * It has to be called before using the service.
+   */
+  public static init(region: string, tableName: string): void {
+    if (!this._instance) {
+      new FeatureFlags(region, tableName);
+    }
   }
 
-  const ddb = makeDdbClient(region, tableName);
-  try {
-    const config = await ddb._docClient
-      .query({
-        TableName: ddb._table,
-        KeyConditionExpression: `${partitionKey} = :id`,
-        ExpressionAttributeValues: { ":id": recordId },
-        ScanIndexForward: false,
-        Limit: 1,
-      })
-      .promise();
-    const record = ddbItemToDbRecord(config.Items?.[0]);
-    if (!record) return undefined;
-
-    featureFlagsCache = { record, timestamp: now };
-    log(`Updated feature flags cache, version: ${record.version}, age was ${age} millis`);
-
-    return record;
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  } catch (error: any) {
-    if (error.code === "ResourceNotFoundException") {
-      return undefined;
+  private static get instance(): FeatureFlags {
+    if (!FeatureFlags._instance) {
+      throw new MetriportError("Feature flags not initialized");
     }
-    throw error;
+    return FeatureFlags._instance;
   }
-}
 
-export async function updateFeatureFlagsRecord({
-  region,
-  tableName,
-  newRecordData,
-  skipVersionCheck = false,
-}: {
-  region: string;
-  tableName: string;
-  newRecordData: FeatureFlagsRecordUpdate;
-  skipVersionCheck?: boolean;
-}): Promise<FeatureFlagsRecord> {
-  ffDatastoreSchema.parse(newRecordData.featureFlags);
-
-  try {
-    const existingRecord = await getFeatureFlagsRecord({ region, tableName, skipCache: true });
-
-    if (
-      !skipVersionCheck &&
-      existingRecord &&
-      existingRecord.version !== newRecordData.existingVersion
-    ) {
-      throw new BadRequestError(`FFs out of sync, reload and try again`, undefined, {
-        existingVersion: existingRecord.version,
-        updateVersion: newRecordData.existingVersion,
-      });
+  public static async getFeatureFlagsRecord({
+    skipCache = false,
+  }: {
+    skipCache?: boolean;
+  } = {}): Promise<FeatureFlagsRecord | undefined> {
+    if (skipCache) await FeatureFlags._getFeatureFlagsRecord();
+    return await FeatureFlags.instance.cache.get(recordId);
+  }
+
+  public static async updateFeatureFlagsRecord({
+    newRecordData,
+    skipVersionCheck = false,
+  }: {
+    newRecordData: FeatureFlagsRecordUpdate;
+    skipVersionCheck?: boolean;
+  }): Promise<FeatureFlagsRecord> {
```

Comment:
Would be nice and probably reduce mistakes to create an abstraction that allows just one kvp in the ff record to be specified, or even specify a payload like

{
  operation: 'add',
  flag: 'TheFlagToUpdate',
  values: [new-cx-uuid-1, new-cx-uuid-2]
}